### PR TITLE
Add TypeScript annotations to MultiSelect

### LINKED ISSUE #12547: Add TypeScript types to MultiSelect

### PR DESCRIPTION
**Pull request by awarrier99**
Closes #12547 

Convert existing propType definitions to TypeScript annotations on the MultiSelect component. This is part of a broader effort to add TypeScript annotations to components, tracked in #12513.

#### Changelog

**Changed**

- `MultiSelect.js` has been converted to `MultiSelect.tsx`
- Added awarrier99 to contributor list
- Changed a `MultiSelect` story to include required prop `label`
- Changed a `MultiSelect` test to return an empty span instead of an empty string for `itemToElement` (a string is not valid as a JSX constructor, which is how the prop is used in the component)
- Changed `MultiSelect` sorting options to use `locale` prop, rather than hardcoded 'en'

#### Testing / Reviewing

- In the `packages/react` directory, run `yarn build`, `yarn storybook`. In the storybook, navigate to `MultiSelect` and confirm that the component's behavior remains unchanged.
- In `MultiSelect.stories.js` and `MultiSelect-test.js`, add `// @ts-check` as the first line in the file (temporarily, for the sake of this test). Confirm that the files do not have compilation errors (related to the `MultiSelect` component). Make changes to the components in those files - change number to string, etc - and confirm that compilation errors appear.

### CODE REVIEW

**Comment by awarrier99** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by andreancardona** [comment]
@awarrier99 Thank you so much for the contribution! Do you mind resolving the merge conflicts please :)

**Comment by awarrier99** [comment]
> @awarrier99 Thank you so much for the contribution! Do you mind resolving the merge conflicts please :)

Sure, they should be resolved now!

**Comment by awarrier99** [comment]
> @awarrier99 Thank you so much for the contribution! Do you mind resolving the merge conflicts please :)

@andreancardona Sure, they should be resolved now!

File: packages/react/src/components/MultiSelect/MultiSelect.tsx
Comment (tw15egan) [inline packages/react/src/components/MultiSelect/MultiSelect.tsx:275]: What's the reasoning for removing this prop?

Comment (awarrier99) [reply packages/react/src/components/MultiSelect/MultiSelect.tsx:275]: It appears that this prop is not actually used by the `ListBox` component, I visually confirmed in the file that no such prop existed and I was also receiving a TypeScript error for passing a non-existent prop

Comment (awarrier99) [reply packages/react/src/components/MultiSelect/MultiSelect.tsx:275]: Although I suppose that prop could've been contained in the `...rest` group of props in the `ListBox` component which is passed to the underlying `div`, in which case I think a change needs to be made explicitly support the prop (or add an index signature), as the only way to suppress the error is with a `@ts-ignore`

Comment (tw15egan) [reply packages/react/src/components/MultiSelect/MultiSelect.tsx:275]: Thanks for clarifying. I'm not seeing any behavioral changes with removing it, so I'll defer to you on TS best practices and what you would like to do here 👍🏻

**Review by tw15egan — APPROVED** [review]
LGTM once the `all-contributors` merge conflict is fixed! 👍 ✅

File: packages/react/src/components/MultiSelect/MultiSelect.tsx
Comment (awarrier99) [reply packages/react/src/components/MultiSelect/MultiSelect.tsx:275]: No problem, and sounds good - I think if the prop isn't being used for anything meaningful it would make the most sense to just remove it rather than still pass it while telling TypeScript to ignore it (just for the `ListBox` component, it's still used in other places in this component)

**Comment by awarrier99** [comment]
> LGTM once the `all-contributors` merge conflict is fixed! 👍 ✅

Resolved!

**Comment by awarrier99** [comment]
@andreancardona @tw15egan I noticed with the previous run of the workflows that there was an error with a snapshot saying that the public API shouldn't be changed without a new semver bump - do I need to do something to resolve that?

**Comment by tw15egan** [comment]
@awarrier99 usually means snapshots need to be updated due to the changes made. If the changes are intentional (in this case, yes we are modifying the component footprint quite a bit), you can update snapshots with `yarn test -u`

**Comment by awarrier99** [comment]
> @awarrier99 usually means snapshots need to be updated due to the changes made. If the changes are intentional (in this case, yes we are modifying the component footprint quite a bit), you can update snapshots with `yarn test -u`. None of these changes should be breaking so we are good to go 👍🏻

Got it, sounds good! I'll leave the snapshots as is then

**Comment by awarrier99** [comment]
@tw15egan sorry I misunderstood, I thought you were saying we were good to go without needing to update the snaps - thanks for taking care of that

**Comment by tw15egan** [comment]
@awarrier99, no worries, I could have worded it better! I just meant it was okay to update them because we are knowingly making changes to the API 👍🏻 Thanks for contributing 🎉

**Review by andreancardona — APPROVED** [review]

File: packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap
Comment (tw15egan) [inline packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap]: @awarrier99 circling back, is there a way to retain the `downshiftProps` here? Looking at the Dropdown ts conversion, I do not see any snapshot changes to the `downshiftProps`. I'm trying to parse the difference in implementation to see what may be causing it 🤔 

Reference PR: https://github.com/carbon-design-system/carbon/pull/13022/files

**Review by tw15egan — CHANGES_REQUESTED** [review]

File: packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap
Comment (tw15egan) [reply packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap]: The other snapshot changes are intentional and are fine 👍🏻

Comment (awarrier99) [reply packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap]: Sure, let me take a look at this

Comment (awarrier99) [reply packages/react/__tests__/__snapshots__/PublicAPI-test.js.snap]: @tw15egan this should be resolved now - I had changed the prop type to match what was in the `Dropdown` component originally, but I've now reverted that change

**Review by tw15egan — APPROVED** [review]

**Pull request merged**
